Feature/issue 151

ava57r opened the pull request.

ava57r committed:
Добавлены структуры FileHeader + BlockHeader

fix #151

ava57r committed:
Добавлено немного константности

ava57r committed:
Добавил метод stBlockHeader::create

ava57r commented:
#151

dmpas commented on src/tool1cd/cfapi/APIcfBase.cpp:
```diff
@@ -8,3 +9,41 @@ unsigned WindowsTickToUnixSeconds(long long windowsTicks)
+	sprintf(buf, "%08x", next_page_addr);
+	strncpy(BlockHeader.next_page_addr_hex, buf, 8);
+
+	return BlockHeader;
```
Я в своё время не смог сообразить, может, ты скажешь, есть тут способ написать это же без сишных функций? или ну его?

ava57r replied:
Я сделал так, вот думал это убрать как раз
```c++ 
std::string hex_prevempty = to_hex_string(prevempty, false);
 			std::copy(hex_prevempty.begin(),
 						hex_prevempty.end(),
						std::begin(block_header.next_page_addr_hex));
```

dmpas replied:
да, похоже на правду. можно даже тест на это сделать.

dmpas reviewed (changes requested):
убрать сишный код. Остальное хорошо.

ava57r committed:
Убраны функции си

Для заполнения полей структуры использованы алгоритмы стандартной
библиотеки.
Добавлен тест для преобразования BlockHeader -> char*
